src/pretix/base/services/mail.py

```diff
@@ -332,3 +340,75 @@ def render_mail(template, context):
         tpl = get_template(template)
         body = tpl.render(context)
     return body
+
+
+def replace_images_with_cid_paths(body_html):
+    if body_html:
+        email = BeautifulSoup(body_html, "lxml")
+        image_counter = 1
+        cid_images = []
+        for image in email.findAll('img'):
+            cid_id = "image_%s" % (image_counter)
+            image_counter = image_counter + 1
+            original_image_src = image['src']
+
+            image['src'] = "cid:%s" % (cid_id)
+
+            cid_images.append({
+                'src': original_image_src,
+                'cid_id': cid_id
+            })
+
+        return email.prettify(), cid_images
+    else:
+        return body_html, []
+
+
+def attach_cid_images(msg, cid_images, verify_ssl=True):
+    if cid_images and len(cid_images) > 0:
+
+        msg.mixed_subtype = 'related'
+        for image in cid_images:
+            try:
+                mime_image = convert_image_to_cid(
+                    image['src'], image['cid_id'], verify_ssl)
+                if mime_image:
+                    msg.attach(mime_image)
+            except Exception as e:
+                print("ERROR attaching CID image %s[%s] %s" % (image['cid_id'], image['src'], str(e)))
+
+    return msg
+
+
+def convert_image_to_cid(image_src, cid_id, verify_ssl=True):
+    try:
+        image_src_split = image_src.split('data:image/png;base64,')
+        if len(image_src_split) == 2:
+            mime_image = MIMEImage(image_src_split[1], _subtype="png", _encoder=encode_noop)
+            mime_image.add_header('Content-Transfer-Encoding', 'base64')
+        else:
+            image_src = normalize_image_url(image_src)
+
+            path = urlparse(image_src).path
+            guess_subtype = os.path.splitext(path)[1][1:]
+
+            response = requests.get(image_src, verify=verify_ssl)
+            mime_image = MIMEImage(
+                response.content, _subtype=guess_subtype)
+
+        mime_image.add_header('Content-ID', '<%s>' % cid_id)
+
+        return mime_image
+    except Exception as e:
+        print("ERROR creating mime_image %s[%s] %s" % (cid_id, image_src, str(e)))
+        return None
+
+
+def normalize_image_url(url):
+    if '//' not in url.lower():
+        if settings.STATIC_URL.startswith('http'):
+            url = "%s%s" % (settings.MEDIA_ROOT, url)
+        else:
+            url = "%s%s%s" % (settings.SITE_URL, settings.STATIC_URL, url)
```

Comment:
I wrote a review comment that this function is fundamentally wrong, just to then realize it is not. Twice. However, there are still some subtleties here and I think we should add a comment here for future readers, since it's really not trivial. Also, there's the following things to consider:

* Please use [urljoin](https://docs.python.org/3/library/urllib.parse.html?highlight=urljoin#urllib.parse.urljoin) to concatenate URLs, to avoid ending up with double-slashes or even broken URLs.
* ``'//' not in url.lower()`` and ``.startswith('http')`` are kinda two ways to do the same thing within two lines, and neither of them is really bullet-prof. I suggest ``'://' not in url`` for both places.
* ``MEDIA_ROOT`` is an internal filesystem path, we need ``MEDIA_URL``
* If this *is* a static file, ``STATIC_URL`` is probably already included in the path.

Here's a proposal:

```
def normalize_image_url(url):
    if '://' not in url:
        """
        If we see a relative URL in an email, we can't know if it is meant to be a media file
        or a static file, so we need to guess. If it is a static file included with the
        ``{% static %}`` template tag (as it should be), then ``STATIC_URL`` is already prepended.
        If ``STATIC_URL`` is absolute, then ``url`` should already be absolute and this
        function should not be triggered. Thus, if we see a relative URL and ``STATIC_URL`` 
        is absolute *or* ``url`` does not start with ``STATIC_URL``, we can be sure this
        is a media file (or a programmer error …).

        Constructing the URL of either a static file or a media file from settings is still
        not clean, since custom storage backends might very well use more complex approaches
        to build those URLs. However, this is good enough as a best-effort approach. Complex
        storage backends (such as cloud storages) will return absolute URLs anyways so this
        function is not needed in that case.
        """
        if '://' not in settings.STATIC_URL and url.startswith(settings.STATIC_URL):
            url = urljoin(settings.SITE_URL, url)
        else:
            url = urljoin(settings.MEDIA_URL, url)
    return url
```